common/sgx/edl/attestation.edl

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/*
+**==============================================================================
+**
+** sgx/attestation.edl:
+**
+**     Internal ECALLs/OCALLs to be used by liboehost/liboecore for SGX-specific
+**     attestation.
+**
+**==============================================================================
+*/
+
+enclave
+{
+    include "openenclave/bits/types.h"
+
+    // Most types in sgxtypes.h are packed to ensure the memory layout is
+    // the same as data retrieved from the CPU. Since EDL does not support
+    // packed structures, these types cannot be defined in EDL. They do not
+    // need the benefit of deep copy because they are all laid out in
+    // contiguous memory.
+    include "openenclave/bits/sgx/sgxtypes.h"
```

Comment:
Suggestion:
We have used opaque types (for lack of a better word) to prevent type information from leaking to code that does not need it. For example,`oe_cert_t` is defined to have enough space to store the certificate  of different crypto libraries.
https://github.com/openenclave/openenclave/blob/344cc4401bbc0abc85994ed16455a3f73516788a/include/openenclave/internal/crypto/cert.h#L17-L21
You can use `oe_cert_t` in code without knowing whether OpenSSLl or mbedTLS is used underneath.

Similarly the following type would avoid `sgxtypes.h` from having to be installed:
```
typedef _oe_sgx_report {
    uint64_t impl[64]; // 64 or whatever the accurate count is
} oe_sgx_report_t;
```

Reply:
I understand the sentiment, but these are structures that are defined by the SGX spec, not by us, so they won't change. In the Intel SDK these are published structures.

Reply:
The rationale for the suggestion was not to safeguard against change, but to avoid publishing sgxtypes.h.
But I agree, it is not that important.